transformation/plugins/org.jboss.tools.fuse.transformation.core/src/org/jboss/tools/fuse/transformation/core/camel/CamelConfigBuilder.java

```diff
@@ -171,21 +173,30 @@ public AbstractCamelModelElement getDataFormat(String id) {
     }
 
     public Collection<AbstractCamelModelElement> getDataFormats() {
-    	return getModel().getCamelContext().getDataformats().values();
+    	if (getModel().getRouteContainer() instanceof CamelContextElement) {
+    		return ((CamelContextElement)getModel().getRouteContainer()).getDataformats().values();
+    	}
+    	return new ArrayList<AbstractCamelModelElement>();
```

Comment:
diamond operator can be used: return new ArrayList<>();
I usually use Collections.emptyList() , not sure what is the best